### PR TITLE
Simplify HanaSR at AZURE_FENCE_AGENT_CONFIGURATION

### PR DESCRIPTION
**Pull request by mpagot**
Simplify code by making AZURE_FENCE_AGENT_CONFIGURATION mandatory when using native fencing with the Azure cloud provider. This removes the need to manage default value consistency across multiple test modules and libraries.

- Related ticket: https://jira.suse.com/browse/TEAM-9764

# Verification run:

## MSI

- sle-15-SP5-HanaSr-Azure-Payg-x86_64-Build15-SP5_2024-12-17T03:03:23Z-hanasr_azure_test_msi@az_Standard_E4s_v3 -> http://openqaworker15.qa.suse.cz/tests/307710 :green_circle: here where the `fence_azure_arm` command is composed http://openqaworker15.qa.suse.cz/tests/307710#step/Verify_azure_fence_agent_MSI/88

 - sle-15-SP3-HanaSr-Azure-Payg-x86_64-Build15-SP3_2024-12-17T03:03:23Z-hanasr_azure_test_msi@az_Standard_E4s_v3 -> http://openqaworker15.qa.suse.cz/tests/307711 :green_circle:  command composition is in http://openqaworker15.qa.suse.cz/tests/307711#step/Verify_azure_fence_agent_MSI/181

## SPN
 - sle-15-SP3-HanaSr-Azure-Payg-x86_64-Build15-SP3_2024-12-17T03:03:23Z-hanasr_azure_test_spn@az_Standard_E4s_v3 -> http://openqaworker15.qa.suse.cz/tests/307712 :green_circle: here where command is composed http://openqaworker15.qa.suse.cz/tests/307712#step/Verify_azure_fence_agent_SPN/96

### CODE REVIEW

**Comment by Amrysliu** [comment]
The code looks good to me. But it also depends on the modification on Job group. So I will wait for the VR. Thanks.

**Comment by mpagot** [comment]
> The code looks good to me. But it also depends on the modification on Job group. So I will wait for the VR. Thanks.

JobGroup side is completed and merged. Yes VR are missing, please do not merge before them are done

**Review by jankohoutek — APPROVED** [review]
LGTM Approved if VR finished successfully

**Pull request merged**
